Test/allocated1/dafny0/Basics.dfy.expect

```diff
@@ -9,14 +9,14 @@ Basics.dfy(148,15): Error: assertion might not hold
 Basics.dfy(167,9): Error: when left-hand sides a[b[18]] and a[4] refer to the same location, they must be assigned the same value
 Basics.dfy(183,9): Error: when left-hand sides x.f and y.f refer to the same location, they must be assigned the same value
 Basics.dfy(195,18): Error: assertion might not hold
-Basics.dfy(197,9): Error: assignment may update an object not in the enclosing context's modifies clause
 Basics.dfy(197,9): Error: target object may be null
+Basics.dfy(197,9): Error: assignment may update an object not in the enclosing context's modifies clause
 Basics.dfy(202,11): Error: left-hand sides m.x and m.x may refer to the same location
 Basics.dfy(213,14): Error: assertion might not hold
 Basics.dfy(273,9): Error: when left-hand sides d.x and c.x refer to the same location, they must be assigned the same value
 Basics.dfy(463,11): Error: assertion might not hold
 Basics.dfy(474,18): Error: assertion might not hold
 Basics.dfy(476,11): Error: assertion might not hold
-Basics.dfy(13,13): Error: variable 'g', which is subject to definite-assignment rules, might be used before it has been assigned
+Basics.dfy(13,13): Error: variable 'g', which is subject to definite-assignment rules, might not have been defined before it is used
```

Comment:
For me they are different: 

```
var x; // Declaration
trait T { function f() } : int // Declaration
var x := 1 // Declaration with definition
class C extends T { function f() : int { 0 } } // Declaration
x := 3; // Assignment
```

So assignment is the right word to use here, because it's about mutating a variable.
I also think `initialized` would be a viable alternative.

Reply:
Or cut out a lot fo the text: `variable 'g', which is subject to definite-assignment rules, might be uninitialized here`

Reply:
I like @cpitclaudel's last suggestion. What do you think, @MikaelMayer? @RustanLeino?

Reply:
Indeed I prefer uninitialized !

Reply:
I just pushed a change adopting @cpitclaudel's suggestion.